### ISSUE TITLE
Automatic checking for program updates

### ISSUE THREAD

**Issue by dg0yt** [OpenOrienteering/mapper#344]
**puzzlepaint reported on Sourceforge [[tickets:#344](https://sourceforge.net/p/oorienteering/tickets/344)]:**

To notify users of program updates, an automatic update notification would be nice to have.

Suggestion:
- Create a file http://oorienteering.sourceforge.net/MapperVersionCheck.php, which returns if a version is outdated given the version number.
- Make the program send a request to this script in a background thread at startup, and notify the user if the version is outdated. Do the request only once each day.
- Possibilities for notification:
  - Instantly with a (potentially annoying) popup
  - With a popup at the next program startup (making sure that the user isn't interrupted while using the program)
  - In the "Tip of the day" field
- Possibilities for the update:
  - Open http://oorienteering.sourceforge.net/?page_id=103 in a web browser
  - Let the script provide the webpage to open (but might lead to unwanted location if hacked)
  - Automatic download and execution of installer

For the Android version, this would be unnecessary if the app would be added into Google Play. The same applies to versions distributed over Linux package repositories.

The application already depends on the QtNetwork module for QtSingleApplication, so no further dependencies must be added.

If you have suggestions for improvements, go ahead and comment.

**Comment by dg0yt** [comment]
For the record, we could now query the Github API for releases if there wasn't the HTTPS issue.

**Comment by Symbian9** [comment]
Maybe better simply open webpage in browser? (for securety reasons)

**Comment by dg0yt** [comment]
API URL for latest release: https://api.github.com/repos/OpenOrienteering/mapper/releases/latest
